examples/plot_metrics_with_error.py

```diff
@@ -0,0 +1,176 @@
+from fairlearn.metrics import MetricFrame
+from fairlearn.experimental.enable_metric_frame_plotting import plot_metric_frame
+
+
+data = fetch_openml(data_id=1590, as_frame=True)
+X = pd.get_dummies(data.data)
+y_true = (data.target == '>50K') * 1
+sex = data.data['sex']
+
+# %%
+# Error Metrics
+# -------------
+# We have many different choices for error metrics. In this notebook we'll just be using a
+# Normal approximation interval (symmetric) and a Wilson score interval (asymmetric).
+#
+# https://en.wikipedia.org/wiki/Binomial_proportion_confidence_interval#Wilson_score_interval
+# https://en.wikipedia.org/wiki/Binomial_proportion_confidence_interval#Normal_approximation_interval
+#
+# We aim to create a 95% confidence interval, so we use a `z_score` of 1.959964
+z_score = 1.959964
+digits_of_precision = 4
+
+
+def wilson(p, n, digits=digits_of_precision, z=z_score):
+    """ Returns lower and upper bound """
+    denominator = 1 + z**2/n
+    centre_adjusted_probability = p + z*z / (2*n)
+    adjusted_standard_deviation = np.sqrt((p*(1 - p) + z*z / (4*n)))/np.sqrt(n)
+    lower_bound = (centre_adjusted_probability - z*adjusted_standard_deviation) / denominator
+    upper_bound = (centre_adjusted_probability + z*adjusted_standard_deviation) / denominator
+    return (round(lower_bound, digits), round(upper_bound, digits))
+
+
+def compute_error_metric(metric_value, sample_size, z_score):
+    """ Standard Error Calculation (Binary Classification)
+
+    Assumes infinitely large population,
+    Should be used when the sampling fraction is small.
+    For sampling fraction > 5%, may want to use finite population correction
+    https://en.wikipedia.org/wiki/Margin_of_error
+
+    Note: 
+        Returns absolute error (%)
+    """
+    return z_score*np.sqrt(metric_value*(1.0-metric_value))/np.sqrt(sample_size)
+
+
+def recall_wilson(y_true, y_pred):
+    assert len(y_true) == len(y_pred)
+    tn, fp, fn, tp = confusion_matrix(y_true, y_pred).ravel()
+    bounds = wilson(tp/(tp+fn), tp + fn, digits_of_precision, z_score)
+    return bounds
+
+def recall_normal_err(y_t, y_p):
+    assert len(y_t) == len(y_p)
+    tn, fp, fn, tp = confusion_matrix(y_t, y_p).ravel()
+    error = compute_error_metric(tp/(tp+fn), tp + fn, z_score=z_score)
+    return (error, error)
+
+def accuracy_wilson(y_true, y_pred):
+    assert len(y_true) == len(y_pred)
+    score = accuracy_score(y_true, y_pred)
+    bounds = wilson(score, len(y_true), digits_of_precision, z_score)
+    return bounds
+
+def accuracy_normal_err(y_true, y_pred):
+    assert len(y_true) == len(y_pred)
+    score = accuracy_score(y_true, y_pred)
+    error = compute_error_metric(score, len(y_true), z_score)
+    return (error, error)
+
+
+# %%
+# Classifier
+# ----------
+# Now we use a Decision Tree to make predictions
+#
+
+classifier = DecisionTreeClassifier(min_samples_leaf=10, max_depth=4)
+classifier.fit(X, y_true)
+y_pred = classifier.predict(X)
```

Comment:
@MiroDudik @riedgar-ms @adrinjalali @hildeweerts to what extent do we want to show best practices here? I'm thinking about using predict on the training data, for example. Do we want to use train-test-split or keep the code as simple as possible? I'm leaning towards splitting but open to be convinced either way 🙂

Reply:
Probably should show the split, if only to minimise future copy/paste problems.

Reply:
Adding a line to do train-test split sounds good to me. Moreover, it can nicely showcase the importance of having a large enough test set for the sake of tighter confidence bounds. I'm assuming that in that case we would show error bounds on the test set performance right?

Reply:
Train/test definitely.